### PR TITLE
[v1.24] use container stats from cadvisor for cri-dockerd

### PR DESCRIPTION
**Pull request by kinarashah**
**Issue**
--- 
The main issue around cpu spike with >=v1.24 RKE1 clusters is Kubernetes deprecating internal dockershim with 1.24 and Rancher now using cri-dockerd - this means that all stats are now collected by cri-dockerd talking to docker api directly instead of getting them from cadvisor. The docker calls themselves seem to be the bottleneck, which changes from system to system. (This is confirmed by internal testing and varying results on different user setups). 

https://github.com/rancher/rancher/issues/38816 

**Fix**
--- 
The workaround would be to build our own kubelet that installs docker plugin for cadvisor and continue reading stats the old way.

### CODE REVIEW

**Comment by kinarashah** [comment]
Note: This PR is still marked as WIP because yet to add drone build logic for it. 

But following tags have the linked fix for testing purposes: 
v1.24: https://github.com/rancher/kubernetes/releases/tag/v1.24.10-rc.1-rancher1 
v1.25: https://github.com/rancher/kubernetes/releases/tag/v1.25.6-rc.1-rancher1 

They have been pulled into rancher/hyperkube for `v1.24.10-rancher3` and `v1.25.6-rancher3` respectively.

**Review by snasovich — APPROVED** [review]
Reviewed non-vendor updates commits, looks sane to me

File: Dockerfile.dapper [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM registry.suse.com/bci/golang:1.19
+
+ARG DAPPER_HOST_ARCH
+ARG DEBIAN_FRONTEND=noninteractive
+ENV HOST_ARCH=${DAPPER_HOST_ARCH} ARCH=${DAPPER_HOST_ARCH}
+
+RUN zypper install -y -f docker rsync awk
+COPY --from=docker/buildx-bin:0.10.2 /buildx /usr/libexec/docker/cli-plugins/docker-buildx
+RUN docker buildx version
```
Comment (Oats87) [inline Dockerfile.dapper:9]: testing?

**Review by Oats87 — APPROVED** [review]
looks good to me!

File: Dockerfile.dapper (hunk 1 above)
Comment (kinarashah) [reply Dockerfile.dapper:9]: oh ya, let me remove

**Pull request merged**